src/ucs/datastruct/string_buffer.h

```diff
@@ -35,10 +62,13 @@ UCS_ARRAY_DECLARE_TYPE(string_buffer, size_t, char)
  * ucs_string_buffer_appendf(&strb, "%x%x", 57005, 48879);
  * @endcode
  */
-#define UCS_STRING_BUFFER_FIXED(_var, _buffer) \
-    ucs_string_buffer_t _var = { \
-        UCS_ARRAY_FIXED_INITIALIZER((_buffer), ucs_static_array_size(_buffer)) \
-    }
+#define UCS_STRING_BUFFER_STATIC(_var, _buffer) \
+    UCS_STRING_BUFFER_FIXED(_var, _buffer, ucs_static_array_size(_buffer))
+
+
+#define UCS_STRING_BUFFER_ONSTACK(_var, _capacity) \
+    UCS_STRING_BUFFER_FIXED(_var, UCS_ARRAY_ALLOC_ONSTACK(string_buffer, _capacity), \
```

Comment:
there could be issue in compilation: macro ```UCS_ALLOCA_MAX_SIZE``` is defined in compiler.h file which is not installed with package, but current file (and some other files) are installed (listed in section ```nobase_dist_libucs_la_HEADERS``` of makefile.am)